Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ILLD7YIGPositionCalibration.py

```diff
@@ -0,0 +1,517 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from mantid.simpleapi import *
+from mantid.kernel import Direction, FloatBoundedValidator
+from mantid.api import FileAction, FileProperty, MatrixWorkspaceProperty, MultipleFileProperty, \
+    NumericAxis, Progress, PropertyMode, PythonAlgorithm, WorkspaceProperty
+from datetime import date
+import math
+import numpy as np
+import os
+from xml.dom import minidom
+import xml.etree.ElementTree as ET
+
+
+class ILLD7YIGPositionCalibration(PythonAlgorithm):
+
+    # helper conversions
+    _RAD_2_DEG = 180.0 / np.pi
+    _DEG_2_RAD = 1.0 / _RAD_2_DEG
+    # number of detectors
+    _D7NumberBanks = 3
+    _D7NumberPixels = 132
+    _D7NumberPixelsBank = 44
+    # an approximate universal peak width:
+    _peakWidth = 2.0 # in degrees
```

Comment:
I guess this meant to include `_peakWidth`, `_beamMask1`, and `_beamMask2` rather than the number of pixels? In general, all of these can be inputs to the algorithm, yes. I will add them to the properties.